### PR TITLE
Fixes #325 - Shaarli does not recognize saved links

### LINKED ISSUE #325: Shaarli does not recognize saved links
Hi, I'm running the shaarli version 0.5.1 and when I save a link shaarli does not have problems. But if I want to save the same link again shaarli does not load the tags.. issue here :(

### PR DESCRIPTION
**Pull request by ArthurHoaro**
PHP doesn't seem to autoconvert objects to strings when they're use as array indexes.

Fixes regression introduced in d9d776af19fd0a191f82525991dafbb56e1bcfcb

Note that this feature is pretty light: if URL's strings doesn't match perfectly, it won't be regognize (eg. HTTP/HTTPS).

#325

### CODE REVIEW

**Comment by ArthurHoaro** [comment]
Updated with a bit of cleanup.

File: index.php [hunk 1]
```diff
@@ -1485,51 +1485,57 @@ function renderPage()
...
-            if (empty($title) && parse_url($url,PHP_URL_SCHEME)=='http')
-            {
+            // Get title if it was provided in URL (by the bookmarklet).
+            $title = (empty($_GET['title']) ? '' : $_GET['title'] );
+            // Get description if it was provided in URL (by the bookmarklet). [Bronco added that]
+            $description = (empty($_GET['description']) ? '' : $_GET['description']);
+            $tags = (empty($_GET['tags']) ? '' : $_GET['tags'] );
+            $private = (!empty($_GET['private']) && $_GET['private'] === "1" ? 1 : 0);
+            // If this is an HTTP(s) link, we try go get the page to extract the title (otherwise we will to straight to the edit form.)
+            if (empty($title) && strpos($url->getScheme(), 'http') !== false) {
```
Comment (virtualtam) [inline index.php]: behavior change: metadata retrieval support for HTTPS scheme

=> should be commited separately, or clearly mentioned in the commit message

File: application/Url.php [hunk 2]
```diff
@@ -147,4 +151,14 @@ public function cleanup()
         $this->cleanupFragment();
         return $this->__toString();
     }
+
+    /**
+     * Get URL scheme
+     */
```
Comment (virtualtam) [inline application/Url.php]: missing `@return`

File: application/Url.php [hunk 3]
```diff
@@ -147,4 +151,14 @@ public function cleanup()
...
+    public function getScheme() {
+        if(!isset($this->parts['scheme'])) {
```
Comment (virtualtam) [inline application/Url.php]: Missing space:
```php
if (...) {
```

File: tests/UrlTest.php [hunk 4]
```diff
@@ -151,4 +151,14 @@ public function testCleanupMixedContent()
...
     }
+
+    /**
+     * Test default http scheme.
+     */
+    public function testDefaultScheme() {
+        $url = new Url(self::$baseUrl);
+        $this->assertEquals('http', $url->getScheme());
+        $url = new Url('domain.tld');
+        $this->assertEquals('http', $url->getScheme());
```
Comment (virtualtam) [inline tests/UrlTest.php]: additional check: ensure other schemes are left unchanged (`ftp://`, `git://`, `ssh://`...)

**Comment by ArthurHoaro** [comment]
Thanks for the code review @virtualtam!

**Comment by virtualtam** [comment]
:+1:

**Comment by nicolasdanelon** [comment]
I see this pull request will be merged to master in 0.5.2 version. But this is something critical, i think, what if you release a subversion 0.5.1.1 like debian packages ? ;) :)

**Comment by ArthurHoaro** [comment]
Another solution would be to push #319 to 0.5.3 and release 0.5.2.

**Comment by virtualtam** [comment]
@nicolasdanelon Shaarli's a bit young to have 4-digit release numbers ^_^
@ArthurHoaro +1 for releasing 0.5.2 as-is

**Comment by nicolasdanelon** [comment]
yes, ok. It's too young. So release the new version!! :D hahaha

**Comment by virtualtam** [comment]
To elaborate a bit on versioning:
- we're planning to use [Semantic Versioning](http://semver.org/) (SemVer) starting `v1.0.0`
- SemVer revisions are `MAJOR.MINOR.PATCH`
- pre-release versions are free-form, though a (very) common convention is `0.MINOR.PATCH`
- starting `v0.5.1`, we've introduced a `stable` branch so users won't experience too much trouble caused by changes on `master`
    - though, feedback is greatly appreciated :)

Minor (bugfix) Shaarli releases should pop quickly enough (1-2 weeks) so we won't have to over-release stuff (with the exception of critical security issues).

**Pull request merged**